marlowe-playground-client/src/MarloweEditor/BottomPanel.purs

```diff
@@ -16,28 +16,33 @@ import Halogen.HTML (ClassName(..), HTML, a, div, li, pre, section, text, ul_)
 import Halogen.HTML.Events (onClick)
 import Halogen.HTML.Properties (class_, classes)
 import MarloweEditor.Types (Action(..), BottomPanelView(..), State, _editorErrors, _editorWarnings, _showErrorDetail, contractHasErrors)
-import StaticAnalysis.BottomPanel (analysisResultPane, analyzeButton)
-import StaticAnalysis.Types (_analysisState, isCloseAnalysisLoading, isReachabilityLoading, isStaticLoading)
+import StaticAnalysis.BottomPanel (analysisResultPane, analyzeButton, clearButton)
+import StaticAnalysis.Types (_analysisExecutionState, _analysisState, isCloseAnalysisLoading, isNoneAsked, isReachabilityLoading, isStaticLoading)
 import Text.Parsing.StringParser.Basic (lines)
 
 panelContents :: forall p. State -> BottomPanelView -> HTML p Action
 panelContents state StaticAnalysisView =
   section
     [ classes [ ClassName "panel-sub-header", aHorizontal, Classes.panelContents ]
     ]
-    [ analysisResultPane state
+    [ analysisResultPane SetIntegerTemplateParam state
     , analyzeButton loadingWarningAnalysis enabled' "Analyse for warnings" AnalyseContract
     , analyzeButton loadingReachability enabled' "Analyse reachability" AnalyseReachabilityContract
     , analyzeButton loadingCloseAnalysis enabled' "Analyse for refunds on Close" AnalyseContractForCloseRefund
+    , clearButton (nothingLoading && not noneAskedAnalysis) "Clear" ClearAnalysisResults
     ]
   where
-  loadingWarningAnalysis = state ^. _analysisState <<< to isStaticLoading
+  loadingWarningAnalysis = state ^. _analysisState <<< _analysisExecutionState <<< to isStaticLoading
 
-  loadingReachability = state ^. _analysisState <<< to isReachabilityLoading
+  loadingReachability = state ^. _analysisState <<< _analysisExecutionState <<< to isReachabilityLoading
 
-  loadingCloseAnalysis = state ^. _analysisState <<< to isCloseAnalysisLoading
+  loadingCloseAnalysis = state ^. _analysisState <<< _analysisExecutionState <<< to isCloseAnalysisLoading
 
-  enabled' = not loadingWarningAnalysis && not loadingReachability && not loadingCloseAnalysis && not contractHasErrors state
+  noneAskedAnalysis = state ^. _analysisState <<< _analysisExecutionState <<< to isNoneAsked
+
+  nothingLoading = not loadingWarningAnalysis && not loadingReachability && not loadingCloseAnalysis
+
+  enabled' = nothingLoading && not contractHasErrors state
```

Comment:
should probably rename this now to `analysisEnabled` or something since you now have similar for the clear button.